petpals/messaging/routes.py

```diff
@@ -0,0 +1,193 @@
+        Messages.query.filter_by(recipient_username=current_user.username)
+        .order_by(Messages.time_sent.asc())
+        .all()
+    )
+
+    for message in sender_messages:
+        recent_conversations.append(message)
+        if message.conversation_id not in unique_conversation_ids:
+            unique_conversation_ids.append(message.conversation_id)
+
+    for message in recipient_messages:
+        recent_conversations.append(message)
+        if message.conversation_id not in unique_conversation_ids:
+            unique_conversation_ids.append(message.conversation_id)
+
+    for id in unique_conversation_ids:
+        for message in recent_conversations:
+            if message.conversation_id in unique_conversation_ids:
+                recent_messages.append(message)
+                unique_conversation_ids.remove(message.conversation_id)
+
+    return render_template(
+        'messages.html',
+        recent_messages=recent_messages,
+        current_user=current_user,
+        active_page=active_page,
+    )
+
+
+@router.get('/new')
+def new_message():
+    usernames = []
+    next_conversation_id = 0
+    conversation_id_query = db.session.query(func.max(Messages.conversation_id)).all()
+
+    if conversation_id_query[0][0] != None:
+        next_conversation_id = (
+            int(db.session.query(func.max(Messages.conversation_id)).all()[0][0]) + 1
+        )
+    else:
+        next_conversation_id = 1
+
+    users = User.query.all()
+    messages = Messages.query.all()
+
+    for user in users:
+        if current_user.username != user.username:
+            usernames.append(user.username)
+
+    for message in messages:
+        if current_user.username == message.sender_username:
+            if message.recipient_username in usernames:
+                usernames.remove(message.recipient_username)
+        elif current_user.username == message.recipient_username:
+            if message.sender_username in usernames:
+                usernames.remove(message.sender_username)
+
+    return render_template(
+        'new_message.html',
+        usernames=sorted(usernames),
+        sender=current_user.username,
+        active_page=active_page,
+        conversation_id=next_conversation_id,
+    )
+
+
+@router.post('/new/<recipient>')
+def send_new_message(recipient):
+    data = request.json
+
+    sender_checker = (
+        Messages.query.with_entities(
+            Messages.sender_username,
+            Messages.recipient_username,
+            Messages.conversation_id,
+        )
+        .filter_by(sender_username=data['sender'], recipient_username=data['recipient'])
+        .all()
+    )
+    recipient_checker = (
+        Messages.query.with_entities(
+            Messages.sender_username,
+            Messages.recipient_username,
+            Messages.conversation_id,
+        )
+        .filter_by(sender_username=data['recipient'], recipient_username=data['sender'])
+        .all()
+    )
```

Comment:
Should use User table instead with a simple filter and a `first_or_404`

Reply:
I need all the values T-T